src/main/java/graphql/annotations/dataFetchers/MethodDataFetcher.java

```diff
@@ -118,40 +132,67 @@ private Object[] invocationArgs(DataFetchingEnvironment environment, ProcessingE
         return result.toArray();
     }
 
+    @SuppressWarnings("ConstantConditions")
     private Object buildArg(Type p, GraphQLType graphQLType, Object arg) {
-        if (arg == null) {
+        Optional<Object> optionalArg = Optional.ofNullable(arg);
+        if (!optionalArg.isPresent()) {
             return null;
         }
```

Comment:
This creation of optional is useless and makes the code much more confusing and complex to understand. You just test that optional isPresent, otherwise return null. An optional which always have a value is not an optional anymore.